src/dialects/db2/query-generator-typescript.ts

```diff
@@ -33,4 +36,26 @@ export class Db2QueryGeneratorTypeScript extends AbstractQueryGenerator {
       'ORDER BY NAME;',
     ]);
   }
+
+  removeIndexQuery(tableName: TableNameOrModel, indexNameOrAttributes: string | string[], options: RemoveIndexQueryOptions) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect.name,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        new Set(),
+        options,
+      );
+    }
+
+    let indexName: string;
+    const table = this.extractTableDetails(tableName);
```

Comment:
We can move this into the if-statement so the table is only retrieved when we need it. This holds for the other dialects as well

Reply:
Done